Catnip docs (#353)
configcat/docs: website/docs/targeting/targeting-overview.md

---
id: targeting-overview
title: Targeting Overview
description: Overview of the ConfigCat targeting feature with examples.
---

Using this feature you will be able to set different setting values for different users in your application. Typical use cases are:
- Beta testing
- A/B testing
- Phased rollouts, canary releases

## How does it work?

1. You define a [Targeting rule] on the <a href="https://app.configcat.com" target="_blank">ConfigCat Dashboard</a>.
2. You pass a [User Object] to the ConfigCat SDK in your application.
3. ConfigCat will use the User Object and the Targeting Rule to determine which features are enabled for the user.

## What is what?
*TODO - new screenshot*
![Targeting Overview](../../static/assets/targeting/targeting.png)

### Feature Flag / Setting

A feature flag is a named boolean setting that can be turned ON or OFF. It is also known as a feature toggle, feature switch, feature flipper, conditional feature, etc.

> The setting is a **string**, **integer**, **double** value that can be used to configure your application. It is also known as a configuration value, configuration setting, configuration parameter, etc.

Author: *TODO*
- ff kifejezest hasznaljuk mert az a leggyakobb fajtaja a settingeknek
- ha ffrol beszelunk akkor az a settingekre is vonatkozik

### Targeting Rule

A targeting rule is a collection of conditions (IF / AND items) and values (THEN item). It is used to determine whether a feature flag or setting is enabled or not for a specific user. More about targeting rules [here].

### Condition (IF / AND item)

> A condition is a boolean expression that can be evaluated to true or false. It is used to determine whether a targeting rule is a match or not for a specific user. More about conditions [here].

Author: *TODO*
- ff kifejezest hasznaljuk mert az a leggyakobb fajtaja a settingeknek
- ha ffrol beszelunk akkor az a settingekre is vonatkozik

---
id: targeting-overview
title: Targeting Overview
description: Overview of the ConfigCat targeting feature with examples.
---

Using this feature you will be able to set different setting values for specific users or groups of users in your application. Typical use cases are:
- Beta testing
- A/B testing
- Phased rollouts, canary releases

## How does it work?

1. On the <a href="https://app.configcat.com" target="_blank">ConfigCat Dashboard</a>, you add [Targeting rules] and/or [Percentage options] to your feature flag.
2. In your application, you pass a [User Object] to the ConfigCat SDK.

> 3. The ConfigCat SDK will use the User Object and the rules defined on the Dashboard to determine whether feature should be enabled for the user.

Reviewer: ```suggestion
3. The ConfigCat SDK will use the User Object and the rules defined on the Dashboard to determine whether the feature should be enabled for the user.
```

Note: "the related" would maybe even better.

## What is what?
*TODO - new screenshot*
![Targeting Overview](../../static/assets/targeting/targeting-overview/targeting.png)

### Feature Flag / Setting

A *setting* is a set of rules that yields a **boolean**, **string**, **integer** or **double** value, which can be used to configure your application. It is also known as a configuration value, configuration setting, configuration parameter, etc.

> A *feature flag* is a setting that yields a **boolean** value, that is used to decide whether an application feature should be turned ON or OFF. It is also known as a feature toggle, feature switch, feature flipper, conditional feature, etc.

Reviewer: Could be even more clear and concise this way:

```suggestion
A *feature flag* is a **boolean** setting that is used to decide whether an application feature should be turned ON or OFF. It is also known as a feature toggle, feature switch, feature flipper, conditional feature, etc.
```

:::info

> Throughout this documentation, we will use the term *feature flag* to refer to both feature flags and settings for simplicity, because feature flags are the most common type of settings.

Reviewer: ```suggestion
Throughout this documentation, we generally use the term *feature flag* to refer to both feature flags and settings for simplicity, because feature flags are the most common type of settings.
```

:::

### Targeting Rule

A *targeting rule* consists of two parts: a collection of conditions and a value to serve. In case the conditions are met (the targeting rule *matches*), the value is served to the user. More about targeting rules [here].

### Condition

A *condition* is a logical expression that can be evaluated to true or false. There are three types of conditions: [User condition], [Flag condition (Prerequisite)], [Segment condition]. For a targeting rule to match, all of its conditions must evaluate to true. More about conditions [here].

### Percentage Options

> Percentage options are used to split users into groups that will receive a specific value for a feature flag. The groups are based on the ratio of the percentages. Percentage options are often used for A/B testing or phased rollouts. More about percentage options [here].

Reviewer: Perhaps it's a bit more accurate this way:

```suggestion
Percentage options are used to split users into groups that will receive a specific value for a feature flag. The grouping is done based on the ratio of the percentages. Percentage options are often used for A/B testing or phased rollouts. More about percentage options [here].
```

### "To all users" / "To all other" / "To unidentified" value

> A feature flag always contains a trivial "rule", a simple value, which comes after the actual rules. This value is returned when none of the preceding rules yields a result. Or they serve as a fallback value if a [User Object] is not given.

Reviewer: I think we can omit the second part as the first part includes it implicitly. Also, we don't really want to get into details at this point.

```suggestion
A feature flag always contains a trivial "rule", a simple value, which comes after the actual rules. This value is returned when none of the preceding rules yields a result.
```

### User Object

> A *user object* is a collection of *user attributes* that describe the properties of a user. Using these attributes in the conditions of the targeting rules, you can target specific users. More about the user object [here].

Reviewer: I think "reference" is a key word here as the user attribute name is a reference to the value which is passed to the SDK evaluation logic in the user's application.

```suggestion
A *user object* is a collection of *user attributes* that describe the properties of a user. Referencing these attributes in conditions allows you to define rules for targeting specific users or groups of users. More about the user object [here].
```

## Examples

### Phased rollout / Canary release / Percentage rollout Scenario

#### Context
Our demo company, Whisker Co. is about to release a new feature called `Park Weather Info`. The stakeholders want to make sure that releasing the new feature will go smoothly and it will be received well by the customers.

#### Goal
To make sure that the new feature is working as expected before releasing it to everyone, we initially want to make the feature available to the employees and to `20%` of the customers only.

#### Solution
Let's create a feature flag called `Enable Park Weather Info` with a Targeting Rule that matches the employees at Whisker Co. and Percentage Options set to `20%` `ON` and `80%` `OFF` for the rest of the users (i.e. for the customers).

Here is what such a feature flag looks like on the Dashboard:
![Phased rollout Example](../../static/assets/targeting/targeting-overview/phased-rollout.jpg)

Here is how we get the value of the feature flag in your application:
```js
import * as configcat from 'configcat-js';

// Create the User Object

> let userObject = new configcat.User(

Reviewer 2: ```suggestion
const userObject = new configcat.User(
```
